content/automate/ManageIQ/Infrastructure/VM/Transform/StateMachines/VmImport.class/__class__.yaml

```diff
@@ -0,0 +1,53 @@
+---
```

Comment:
@matobet We're working on documenting guidelines for adding State Machines to the ManageIQ domain. 

Can you modify your ImportVM State Machine as follows:

Minimum Requirements for ManageIQ domain State Machines:
(In addition to the specific processing states required by the new state machine)   

1. Minimum of 3 Pre states(pre1, pre2, pre3)
2. Minimum of 3 Post states(post1, post2, post3)

3. update_*_status method required for every states on_entry, on_exit, on_error
 a. * update request with enhanced message including 
       updated_message  = "[#{$evm.root['miq_server'].name}] "
       updated_message += "Step [#{$evm.root['ae_state']}] "
       updated_message += "Status [#{status}] "
       updated_message += "Message [#{prov.message}] "
       updated_message += "Current Retry Number [#{$evm.root['ae_state_retries']}]" if 
            $evm.root['ae_result'] == 'retry'
 b. * set user_message in request:
       prov.miq_request.user_message = updated_message

 c. * create notification and error log message on error
      if $evm.root['ae_result'] == "error"
          $evm.create_notification(:level => "error" \
                            :message => "VM Provision Error: #{updated_message}")
          $evm.log(:error, "VM Provision Error: #{updated_message}")
      end

4. Finished state 
    Can use CommonMethods finished method: 
    /System/CommonMethods/StateMachineMethods/your_instance_name_here
    (Add an instance to the /System/CommonMethods/StateMachineMethods class with your      instance name. Can optionally modify task_finished method in that class for special processing)